### PR TITLE
break apart reading/homework task plans, add nudge stats

### PR DESCRIPTION
**Pull request by nathanstitt**
This way the assignment will use same format as task completion

### CODE REVIEW

File: app/subsystems/stats/calculations/exercises.rb [hunk 1]
```diff
@@ -10,6 +10,17 @@ def exec(interval:)
                tasked_type: "Tasks::Models::Tasked#{task_type.classify}")
         .count
     end
-  end
 
+
+    nudge_stats = Tasks::Models::TaskedExercise
+      .where('response_validation is not null')
```
Comment (Dantemss) [inline app/subsystems/stats/calculations/exercises.rb]: Might wanna exclude `'{}'` too? Though I'm not 100% sure it happened in the data.

File: app/subsystems/stats/calculations/exercises.rb [hunk 2]
```diff
@@ -10,6 +10,17 @@ def exec(interval:)
...
+      .where(:updated_at => interval.range)
+      .select(
+      "coalesce(sum((case when jsonb_array_length(response_validation->'attempts') > 0 then 1 else 0 end)), 0) as nudge_calculated,
+      coalesce(sum((case when (response_validation->'attempts'->0->'valid') = 'true' then 0 else 1 end)), 0) as nudge_initially_invalid,
+      coalesce(sum((case when (response_validation->'attempts'->1->'valid') = 'true' then 1 else 0 end)), 0) as nudge_retry_correct"
```
Comment (Dantemss) [inline app/subsystems/stats/calculations/exercises.rb]: Oh I guess the coalesce is for `'{}'`?

File: app/subsystems/stats/calculations/exercises.rb [hunk 3]
```diff
@@ -10,6 +10,17 @@ def exec(interval:)
...
+    ).all[0].as_json(except: :id)
```
Comment (Dantemss) [inline app/subsystems/stats/calculations/exercises.rb]: Maybe it's easier to return the whole json and process in the FE? Considering the FE is the one that makes it...

File: app/subsystems/stats/calculations/exercises.rb (hunk 1 above)
Comment (nathanstitt) [reply app/subsystems/stats/calculations/exercises.rb]: Yeah, I'm not sure either.  I'll update to filter those also

File: app/subsystems/stats/calculations/exercises.rb (hunk 2 above)
Comment (nathanstitt) [reply app/subsystems/stats/calculations/exercises.rb]: nah, that's for the case where there were no rows found, in which case `sum` will return `null`

File: app/subsystems/stats/calculations/exercises.rb (hunk 3 above)
Comment (nathanstitt) [reply app/subsystems/stats/calculations/exercises.rb]: probably not the best idea since were currently at 63,000 or so results 😁

**Pull request merged**